spec/requests/providers/proceeding_loop/emergency_level_of_service_spec.rb

```diff
@@ -0,0 +1,100 @@
+require "rails_helper"
+
+RSpec.describe "EmergencyLevelOfServiceController", :vcr, type: :request do
+  let(:application) do
+    create(:legal_aid_application,
+           :with_proceedings,
+           :with_delegated_functions_on_proceedings,
+           explicit_proceedings: %i[da001],
+           df_options: { DA001: [10.days.ago, 10.days.ago] },
+           substantive_application_deadline_on: 10.days.from_now)
+  end
+  let(:application_id) { application.id }
+  let(:proceeding_id) { application.proceedings.first.id }
+  let(:provider) { application.provider }
+
+  before do
+    allow(Setting).to receive(:enable_mini_loop?).and_return(true) # TODO: Remove when the mini-loop feature flag is removed
+    allow(Setting).to receive(:enable_loop?).and_return(true) # TODO: Remove when the loop feature flag is removed
+  end
+
+  describe "GET /providers/applications/:legal_aid_application_id/emergency_level_of_service/:proceeding_id" do
+    subject(:get_elos) { get "/providers/applications/#{application_id}/emergency_level_of_service/#{proceeding_id}" }
+
+    context "when the provider is not authenticated" do
+      before { get_elos }
+
+      it_behaves_like "a provider not authenticated"
+    end
+
+    context "when the provider is authenticated" do
+      before do
+        login_as provider
+        get_elos
+      end
+
+      it "returns http success" do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "displays the proceeding header" do
+        expect(response.body).to include("Proceeding 1")
+        expect(response.body).to include("Inherent jurisdiction high court injunction")
+      end
+
+      context "when the proceeding has only one possible level of service" do
+        it "displays a message that the level of service cannot be changed" do
+          expect(response.body).to include("You cannot change the default level of service for the emergency application for this proceeding")
+        end
+      end
+
+      context "when the proceeding has more than one possible level of service" do
+        let(:application) do
+          create(:legal_aid_application,
+                 :with_proceedings,
+                 :with_delegated_functions_on_proceedings,
+                 explicit_proceedings: %i[se013],
+                 df_options: { SE013: [10.days.ago, 10.days.ago] },
+                 substantive_application_deadline_on: 10.days.from_now)
+        end
+
+        it "displays a message that the level of service cannot be changed" do
+          expect(response.body).to include("For the emergency application, select the level of service")
+        end
+      end
+    end
+  end
+
+  describe "POST /providers/applications/:legal_aid_application_id/emergency_level_of_service/:proceeding_id" do
+    subject(:post_sd) { patch "/providers/applications/#{application_id}/emergency_level_of_service/#{proceeding_id}", params: }
```

Comment:
Minor nitpick, can you rename the subject to `post_elos` so it matches the get variable above?

Reply:
👍 have udpated